src/Orleans/Runtime/AsynchQueueAgent.cs

```diff
@@ -30,7 +30,10 @@ public void QueueRequest(T request)
                 queueTracking.OnEnQueueRequest(1, requestQueue.Count, request);
             }
 #endif
-            requestQueue.Add(request);
+            if (requestQueue != null)
+            {
+                requestQueue.Add(request);
```

Comment:
Simply for understandability - the null check is odd, so the comment mentions when it might be null.